src/FffGcodeWriter.cpp

```diff
@@ -3714,12 +3756,15 @@ void FffGcodeWriter::setExtruder_addPrime(const SliceDataStorage& storage, Layer
 
 void FffGcodeWriter::addPrimeTower(const SliceDataStorage& storage, LayerPlan& gcode_layer, const size_t prev_extruder) const
 {
-    if (! Application::getInstance().current_slice_->scene.current_mesh_group->settings.get<bool>("prime_tower_enable"))
+    if (! storage.primeTower.enabled_)
     {
         return;
     }
 
-    storage.primeTower.addToGcode(storage, gcode_layer, prev_extruder, gcode_layer.getExtruder());
+    LayerIndex layer_nr = gcode_layer.getLayerNr();
+    const std::vector<ExtruderUse>& extruder_order
+        = (layer_nr < 0) ? extruder_order_per_layer_negative_layers[extruder_order_per_layer_negative_layers.size() + layer_nr] : extruder_order_per_layer[layer_nr];
```

Comment:
I'm a bit puzzled by this line; Am I correct to assume that the negative (raft) layers are appended to the end of the `extruder_order_per_layer_negative_layers` vec; e.g. if we were to map the values of the `extruder_order_per_layer_negative_layers` to the layernr we would get something like this (assume we have 1000 layers)
```
map(layer => layer.layer_nr,extruder_order_per_layer_negative_layers) == [0, 1, 2, 3, ... 998, 999, 1000, -3, -2, -1]
```

Reply:
No they are not appended, we have two different vectors: `extruder_order_per_layer_negative_layers` which contains only the raft layers, and `extruder_order_per_layer` which contains only the model layers. This was actually the previously existing behavior.